File: backend/src/main/java/com/zhiqin/recruitment/entity/Position.java
package com.zhiqin.recruitment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("position")
public class Position implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField("title")
    private String title;

    @TableField("enterprise_id")
    private Long enterpriseId;

    @TableField("enterprise_name")
    private String enterpriseName;

    @TableField("category")
    private String category;

    @TableField("salary_min")
    private Integer salaryMin;

    @TableField("salary_max")
    private Integer salaryMax;

    @TableField("city")
    private String city;

    @TableField("education")
    private String education;

    @TableField("experience")
    private String experience;

    @TableField("responsibility")
    private String responsibility;

    @TableField("requirement")
    private String requirement;

    @TableField("view_count")
    private Integer viewCount;

    @TableField("apply_count")
    private Integer applyCount;

    @TableField("expire_time")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    private LocalDateTime expireTime;

    @TableField("status")
    private Integer status;

    @TableField("create_time")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    private LocalDateTime createTime;

    @TableField("update_time")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    private LocalDateTime updateTime;

}

File: frontend/src/api/position.ts
import request from '@/utils/request'

export interface PositionQuery {
  pageNum: number
  pageSize: number
  title?: string
  enterpriseId?: number | string
  category?: string
  city?: string
  status?: number | string
}

export interface PositionForm {
  id?: number
  title: string
  enterpriseId: number | string
  category: string
  salaryMin: number
  salaryMax: number
  city: string
  education?: string
  experience?: string
  responsibility?: string
  requirement?: string
  expireTime?: string
  status?: number | string
}

export interface PositionRecord {
  id: number
  title: string
  enterpriseId: number
  enterpriseName: string
  category: string
  salaryMin: number
  salaryMax: number
  city: string
  education: string
  experience: string
  responsibility: string
  requirement: string
  viewCount: number
  applyCount: number
  status: number
  expireTime: string
  createTime: string
  updateTime: string
}

export interface ResumeRecordVO {
  id: number
  seekerId: number
  seekerName: string
  positionTitle: string
  enterpriseName: string
  status: number
  statusText: string
  remark: string
  applyTime: string
}

export interface PositionDetailVO extends PositionRecord {
  resumeList: ResumeRecordVO[]
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export function getPositionList(params: PositionQuery) {
  return request<PageResult<PositionRecord>>({
    url: '/position/list',
    method: 'get',
    params
  })
}

export function getPositionDetail(id: number) {
  return request<PositionRecord>({
    url: `/position/${id}`,
    method: 'get'
  })
}

export function getPositionDetailVO(id: number) {
  return request<PositionDetailVO>({
    url: `/position/detail/${id}`,
    method: 'get'
  })
}

export function createPosition(data: PositionForm) {
  return request<void>({
    url: '/position',
    method: 'post',
    data
  })
}

export function updatePosition(data: PositionForm) {
  return request<void>({
    url: `/position/${data.id}`,
    method: 'put',
    data
  })
}

export function removePosition(id: number) {
  return request<void>({
    url: `/position/${id}`,
    method: 'delete'
  })
}

export function removePositionBatch(ids: number[]) {
  return request<void>({
    url: '/position/batch',
    method: 'delete',
    data: { ids }
  })
}

export function updatePositionStatus(ids: number[], status: number) {
  return request<void>({
    url: '/position/status',
    method: 'put',
    data: { ids, status }
  })
}

export function onlinePosition(id: number) {
  return request<void>({
    url: `/position/online/${id}`,
    method: 'put'
  })
}

export function offlinePosition(id: number) {
  return request<void>({
    url: `/position/offline/${id}`,
    method: 'put'
  })
}
